pkg/build/controller/strategy/sti.go

```diff
@@ -76,31 +77,103 @@ func (bs *SourceBuildStrategy) CreateBuildPod(build *buildapi.Build) (*v1.Pod, e
 			ServiceAccountName: build.Spec.ServiceAccount,
 			Containers: []v1.Container{
 				{
-					Name:  "sti-build",
-					Image: bs.Image,
-					Env:   containerEnv,
+					Name:    "sti-build",
+					Image:   bs.Image,
+					Command: []string{"openshift-sti-build"},
+					Env:     containerEnv,
 					// TODO: run unprivileged https://github.com/openshift/origin/issues/662
 					SecurityContext: &v1.SecurityContext{
 						Privileged: &privileged,
 					},
-					Args: []string{},
 					TerminationMessagePolicy: v1.TerminationMessageFallbackToLogsOnError,
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      "buildworkdir",
+							MountPath: buildutil.BuildWorkDirMount,
+						},
+					},
+					ImagePullPolicy: v1.PullIfNotPresent,
+					Resources:       buildutil.CopyApiResourcesToV1Resources(&build.Spec.Resources),
+				},
+			},
+			Volumes: []v1.Volume{
+				{
+					Name: "buildworkdir",
+					VolumeSource: v1.VolumeSource{
+						EmptyDir: &v1.EmptyDirVolumeSource{},
+					},
 				},
 			},
 			RestartPolicy: v1.RestartPolicyNever,
 			NodeSelector:  build.Spec.NodeSelector,
 		},
 	}
-	pod.Spec.Containers[0].ImagePullPolicy = v1.PullIfNotPresent
-	pod.Spec.Containers[0].Resources = util.CopyApiResourcesToV1Resources(&build.Spec.Resources)
+
+	if build.Spec.Source.Git != nil || build.Spec.Source.Binary != nil {
+		gitCloneContainer := v1.Container{
+			Name:    "git-clone",
+			Image:   bs.Image,
+			Command: []string{"openshift-git-clone"},
+			Env:     containerEnv,
+			TerminationMessagePolicy: v1.TerminationMessageFallbackToLogsOnError,
+			VolumeMounts: []v1.VolumeMount{
+				{
+					Name:      "buildworkdir",
+					MountPath: buildutil.BuildWorkDirMount,
+				},
+			},
+			ImagePullPolicy: v1.PullIfNotPresent,
+			Resources:       buildutil.CopyApiResourcesToV1Resources(&build.Spec.Resources),
+		}
+		if build.Spec.Source.Binary != nil {
+			gitCloneContainer.Stdin = true
+			gitCloneContainer.StdinOnce = true
+		}
+		pod.Spec.InitContainers = append(pod.Spec.InitContainers, gitCloneContainer)
+	}
+	if len(build.Spec.Source.Images) > 0 {
+		extractImageContentContainer := v1.Container{
+			Name:    "extract-image-content",
+			Image:   bs.Image,
+			Command: []string{"openshift-extract-image-content"},
+			Env:     containerEnv,
+			// TODO: run unprivileged https://github.com/openshift/origin/issues/662
```

Comment:
hm, i guess not, i guess that's a separate scc permission.  but if we don't need it here, i'm not sure why we ever needed to run builds in privileged pods?  even creating+running containers would have been done via the docker socket.